src/ZenohUTransport.cpp

```diff
@@ -108,16 +108,16 @@ v1::UAttributes ZenohUTransport::attachmentToUAttributes(
 	        .deserialize<std::vector<std::pair<std::string, std::string>>>();
 
 	if (attachment_vec.size() != 2) {
-		spdlog::error("attachmentToUAttributes: attachment size != 2");
-		// TODO: error report, exception?
+		throw std::invalid_argument("invalid attachment size");
 	}
 
 	if (attachment_vec[0].second.size() == 1) {
 		if (attachment_vec[0].second[0] != UATTRIBUTE_VERSION) {
-			spdlog::error("attachmentToUAttributes: incorrect version");
-			// TODO: error report, exception?
+			throw std::invalid_argument("incorrect UAttribute version");
 		}
-	};
+	} else {
+		throw std::invalid_argument("incorrect UAttribute version size");
```

Comment:
The error handling model we have been using for up-cpp requires that exceptions only be thrown in a situation where the caller has done something unrecoverable (i.e. passed an invalid UUri to an interface) - exceptions are a panic situation and are not expected to be caught. Faults on the network or with incoming messages should not throw an exception as those faults are transient and the developer of the entity could not have reasonably accounted for this - a bad actor on the network should be ignored.

These should not be exceptions. Maybe we need a way of reporting bad incoming messages to an upper layer, but that would be done in another way (e.g. via callback that receives a UStatus)

Reply:
Zenoh-cpp also can throw the exception if something going wrong (ZException inherited from the std::exception) e.g. from deserialize. So there is nothing wrong with catching them together.
But yes, if we have a hard rule not to produce any exceptions zenoh-cpp allows to pass a parameter with a return code, then check it, pass it to the calling function, and so on. But IMHO it will be more cumbersome.